XIPs/xip-universal-consent-layer.md

```diff
@@ -0,0 +1,55 @@
+# XIP-XX: Universal Consent Layer for XMTP
+
+| XIP                |                                            |
+|--------------------|--------------------------------------------|
+| Title              | Universal Consent Layer for XMTP           |
+| Author             | Pol Maire pol@getconverse.app              |
+| Status             | Draft                                      |
+| Type               | Standards Track                            |
+| Created            | 2023-04-08                                 |
+
+## Abstract
+
+This XIP proposes adding a universal consent layer to the XMTP protocol to prevent spam and allow users to maintain their communication preferences across applications.
+
+## Motivation
+
+Currently, the XMTP network allows anyone to write to anyone without the option to opt-in or opt-out of conversations with other users. Implementing a universal consent layer will provide users with more control over their communication preferences, reduce spam, and enhance privacy.
+
+## Specification
+
+### Consent layers
+
+1. **Opted-in**: The user has either sent at least one message in a conversation with the other user or explicitly accepted talking with the other user.
+2. **Opted-out**: The user has explicitly blocked the other user by clicking on a button in the interface of an XMTP client. Every conversation with the opted-out user is also considered opted-out.
+3. **Neither opted-in nor opted-out**: The user received a message from another user but did not send anything nor clicked on any consent button.
+
+### User Interface
+
+- Clients should create a "requests" section in their user interface containing all conversations with users in the "Neither opted-in or opted-out" category.
+- Clients can add an additional layer of consent, "Implicitly opted-in", at the client level to automatically accept users if there is a good chance of a positive interaction (e.g., having an on-chain transaction with the user or following the user on Lens or Farcaster). This additional layer is not at the protocol level for now.
+
+### User Settings
+
+- User settings do not currently exist at the protocol level. There should be another XIP to address this specific topic.
```

Comment:
One concern we are working through with protocol level user settings is trust. While the settings can be encrypted/signed with the user's keys, if we use the XMTP keys to authenticate the settings it means that a malicious (or growth-minded) client application could modify those settings without the user's knowledge.

For example, apps might "follow" their own support channels in the background, or even sell opt-ins to marketers.

Other, non-malicious, apps would then see these settings as authentic.

The only way around this problem I can see is to require these kinds of setting changes to be signed by the wallet, which does require explicit user consent...but also adds significant friction.